src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java

```diff
@@ -478,7 +479,13 @@ private void processPackage(DetailAST ast, DetailAST nextToken) {
                 log(ast.getLineNo(), MSG_SHOULD_BE_SEPARATED, ast.getText());
             }
         }
-        if (!hasEmptyLineAfter(ast)) {
+        if (hasEmptyLineAfter(ast)) {
+            final Optional<DetailAST> comment = findComment(ast);
+            if (comment.isPresent()) {
+                log(comment.get().getLineNo(), MSG_SHOULD_BE_SEPARATED, comment.get().getText());
```

Comment:
this code become a non-sense for reader:
```
if (hasEmptyLineAfter(ast)) { 
  ... 
  log(comment.get().getLineNo(), MSG_SHOULD_BE_SEPARATE
}
```

we need to update `hasEmptyLineAfter` with comments find.

Reply:
you cannot do that. This method is used in other methods. Also, it will log on ast but it should log on comment token.

Reply:
we shuould do refactoring on names, code should be read as English text and NO cheating.
can you propose something ?

Reply:
Something like 
```
Optional<DetailAST> comment = findComment(ast)
if (comment.isPresent() log comment
else if(!hasEmptyLineAfter(ast) log next sibling
```
?

Reply:
much better !
please do not forget to regenerate diff report.

Reply:
changed, please take a look.

Report will be later